### PR TITLE
Additional snowflake support

### PR DESCRIPTION
**Pull request by create-new-entity**

### CODE REVIEW

**Comment by create-new-entity** [comment]
This changes were made so that queries that have **_or replace_** or some scale/range in **_BINARY, VARBINARY_** types can be parsed as well.

```
create or replace TABLE DELETETHISDB.PUBLIC."2" (
	"att5" TIMESTAMP_TZ(9) NOT NULL,
	"att4" BINARY(8388608),
	primary key ("att5")
);
```

File: pegjs/snowflake.pegjs [hunk 1]
```diff
@@ -508,6 +508,7 @@ create_domain_stmt
     }
 create_table_stmt
   = a:KW_CREATE __
+    orre: (KW_OR __ KW_REPLACE __)?
```
Comment (create-new-entity) [inline pegjs/snowflake.pegjs:511]: CREATE OR REPLACE

File: pegjs/snowflake.pegjs [hunk 2]
```diff
@@ -4741,8 +4744,8 @@ boolean_type
   = t:(KW_BOOL / KW_BOOLEAN) { /* => data_type */ return { dataType: t }}
 
 binary_type
-  = t:(KW_BINARY / KW_VARBINARY) { /* => data_type */ return { dataType: t }; }
-
+  = t:(KW_BINARY / KW_VARBINARY) __ LPAREN __ l:[0-9]+ __ r:(COMMA __ [0-9]+)? __ RPAREN __ s:numeric_type_suffix?  { /* => data_type */ return { dataType: t, length: parseInt(l.join(''), 10), scale: r && parseInt(r[2].join(''), 10), parentheses: true, suffix: s  }; }
+    / t:(KW_BINARY / KW_VARBINARY) { /* => data_type */ return { dataType: t }; }
```
Comment (create-new-entity) [inline pegjs/snowflake.pegjs:4748]: BINARY(2323) or VARBINARY(2323)
BINARY(2323, 2323) or VARBINARY(2323, 2323)

File: pegjs/snowflake.pegjs [hunk 3]
```diff
@@ -4777,7 +4780,7 @@ time_type
   / t:(KW_TIME / KW_TIMESTAMP) __ tz:timezone? { /* =>  data_type */  return { dataType: t, suffix: tz }; }
 
 datetime_type
-  = t:(KW_DATE / KW_DATETIME) __ LPAREN __ l:[0-9]+ __ RPAREN { /* =>  data_type */ return { dataType: t, length: parseInt(l.join(''), 10), parentheses: true }; }
+  = t:(KW_DATE / KW_DATETIME / KW_TIMESTAMP_TZ) __ LPAREN __ l:[0-9]+ __ RPAREN { /* =>  data_type */ return { dataType: t, length: parseInt(l.join(''), 10), parentheses: true }; }
```
Comment (create-new-entity) [inline pegjs/snowflake.pegjs:4783]: Allow TIMESTAMP_TZ

File: pegjs/snowflake.pegjs (hunk 1 above)
Comment (sofia6998) [reply pegjs/snowflake.pegjs:511]: Should it be `or:` for consistency? As here: https://github.com/taozhi8833998/node-sql-parser/blob/d59020b0509fbcabc200b2951165597e90ffa0d7/pegjs/mysql.pegjs#L735

**Review by sofia6998 — APPROVED** [review]
Good findings!

**Pull request merged**